app/models/location.rb

```diff
@@ -7,15 +9,23 @@ class Location < ApplicationRecord
   validates_associated :ips
 
   validates :address, presence: true
+  validates :postcode, presence: true
+  validate :validate_postcode
 
   before_create :set_radius_secret_key
 
   def full_address
-    postcode.blank? ? address : "#{address}, #{postcode}"
+    "#{address}, #{postcode}"
   end
 
 private
 
+  def validate_postcode
+    if postcode.nil? || !UKPostcode.parse(postcode).valid?
```

Comment:
Do we need a nil check?
Won't nil be invalid according to this gem?

Reply:
In the Location model tests it moans about postcode being nil as on create I assume it doesn't  create a postcode from what I'm aware of? @emileswarts

Reply:
So can the first part of this conditional be removed? Not 100% clear on this.

Reply:
It can't

Reply:
```ruby
unless UKPostcode.parse(postcode.to_s).valid?
     errors.add(:postcode, "must be valid")
end
```

Reply:
I LIKE